File: 2364-Count Number of Bad Pairs.py
class Solution:
    def countBadPairs(self, nums: List[int]) -> int:
        # rearrange the formula: nums[i] - i != nums[j] - j
        hmap = dict()
        N = len(nums)
        good = 0
        for i, num in enumerate(nums):
            curr = num - i
            if curr in hmap:
                good += hmap[curr]
            hmap[curr] = hmap.get(curr, 0) + 1
            
        return N*(N-1)//2 - good

File: C++/2364-Count Number of Bad Pairs.cpp
class Solution {
public:
    long long countBadPairs(vector<int>& nums) {
        unordered_map<int, int> cnt;
        int N = nums.size();
        long long res = 0;
        
        for (int i=0; i<N; i++) {
            res += i - cnt[nums[i]-i];
            cnt[nums[i]-i]++;
        }

        return res;
    }
};